samcli/cli/options.py

```diff
@@ -20,8 +20,14 @@ def callback(ctx, param, value):
         state.debug = value
         return value
 
+    # NOTE: --debug option should be eager to be evaluated before other parameters and to set log level to DEBUG
+    # before any other option/parameter processing will require to outbut debug info.
+    # Otherwise parameters are evaluated according to their order and if --debug is specified at the end of the command
+    # some debug output can be lost
+    # https://click.palletsprojects.com/en/7.x/advanced/#callback-evaluation-order
     return click.option(
         "--debug",
+        is_eager=True,
```

Comment:
We are using is_eager in some other options. Can you please check the callback order is correct if more than one `is_eager` options?

Reply:
Yes. The order is okay. There is no problem in this case because `--debug` doesn't do anything but sets the log level.
Here is the list of eager options:
- help, version (by default)
- info
- [config-file](https://github.com/aws/aws-sam-cli/blob/4e1a9796d9538e37dacc170d550c4f5561e10f63/samcli/cli/cli_config_file.py#L244), [config-env](https://github.com/aws/aws-sam-cli/blob/4e1a9796d9538e37dacc170d550c4f5561e10f63/samcli/cli/cli_config_file.py#L265), and [the full block](https://github.com/aws/aws-sam-cli/blob/4e1a9796d9538e37dacc170d550c4f5561e10f63/samcli/cli/cli_config_file.py#L200)
- template-file
- guided